pkg/eval/compile_effect.go

```diff
@@ -497,21 +497,31 @@ type invalidFD struct{ fd int }
 
 func (err invalidFD) Error() string { return fmt.Sprintf("invalid fd: %d", err.fd) }
 
+func (op *redirOp) fileChan() chan interface{} {
```

Comment:
This method only depends on the `mode` field of `redirOp` and there is no other benefit of coupling it with `redirOp`. Turn it into `func fileChan(mode parse.RedirMode) chan interface{}`.